### DISCUSSION TITLE
Issues with install

### DISCUSSION THREAD

**Discussion by nhanch1** [karlcswanson/micboard#45] [Q&A]
I have tried installing on pi using the updated instructions on github but I am having a failure when I try to run micboard ( "npm run build" ). Once I try to run, it shows this:

ERROR in ./node_modules/@shopify/draggable/build/esm/shared/utils/closest/closest.mjs 26:22
Module parse failed: Unexpected token (26:22)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|       return current;
|     }
>     current = current?.parentNode || null;
|   } while (current != null && current !== document.body && current !== document);
|   return null;
 @ ./node_modules/@shopify/draggable/build/esm/Droppable/Droppable.mjs 1:0-58 86:21-28 176:11-18
 @ ./node_modules/@shopify/draggable/build/esm/index.mjs
 @ ./js/dnd.js
 @ ./js/app.js
 @ multi whatwg-fetch ./js/app.js
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! micboard@0.8.5 build: `node node_modules/webpack/bin/webpack.js --progress --hide-modules --config=webpack.config.js`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the micboard@0.8.5 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

Any ideas?
Thank you!

**Comment by dustenharward** [comment]
I would love to be bale to use this application again. such a useful tool! It wont install on anything i have at the moment. I have tried both the Pi and Mac versions and both have outdated extensions that wont install and the newer replacements dont work due to deprecated functions. 

Is there any way we could get a functioning pi image that wont require installation of individual components? if someone can get me a fully functioning Raspberry Pi image I am happy to make a public link to it on my hosting server to make it widely available.

**Reply by s0600204** [reply]
If you don't mind building from source, does the [0.8.7-updates](https://github.com/karlcswanson/micboard/tree/0.8.7-updates) branch work for you?

Personally I can get `micboard` building on a debian "bookworm" system (with `nodejs` from debian's repositories, not `nodesource`), but I don't have any Shure kit to test that what's built functions correctly. (I'm in the process of adding Sennheiser device support, which is why I was trying to get it to build in the first place...)

**Reply by gunner4565** [reply]
I'm having all the same problems as everyone else.  No success on Pi 5 bookworm with any method including Docker.  No luck with 0.8.7-updates either.  What is the command for "(with nodejs from debian's repositories, not nodesource"?????  Might that help?

**Reply by s0600204** [reply]
The installation instructions (for debian systems) contain the following line:

```
curl -sL https://deb.nodesource.com/setup_10.x | sudo -E bash -
```

In case it is unclear, this adds a new location for `apt` to search for node installation packages - one that provides packages that take priority over the default debian packages - and as a side effect ensures that node is installed at a specific version.

In my case, I skipped this and moved onto the next line (`sudo apt-get install nodejs`), thus installing node from debian's repos instead of from nodesource.

If you wish to try this yourself, you'll need to either start afresh or manually undo the pertinent changes that the nodesource script makes (removing `/etc/apt/sources.list.d/nodesource.list` and rerunning `sudo apt-get install nodejs` - making sure it isn't reinstalling `node10` - should do it, but I haven't tested that)

----

Knowing the exact error messages emitted might help diagnose specific problems.

**Reply by gunner4565** [reply]
Thanks much for your help on this...

Fresh Install - The install did go further leaving out the nodesource line.  Here is what I last tried from steinersupport discussion.  Said this is what worked for him. 

sudo apt update
sudo apt-get install git python3-pip
sudo apt-get install -y nodejs
sudo apt-get install gcc g++ make
sudo apt autoremove
sudo apt-get update && sudo apt-get install yarn
sudo apt install npm
sudo apt install python2
wget https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.1/install.sh
bash install.sh
source ~/.bashrc
command -v nvm
nvm install v10.24.1
git clone https://github.com/karlcswanson/micboard.git
cd micboard/
npm install --only=prod     FAILS lots of deprecation errors
pip3 install -r py/requirements.txt  FAILS error: externally managed environment

UNSURE what to do now?? Our church would really like to get this running.
Jerry

**Reply by s0600204** [reply]
Whilst installing `npm` is a step in the right direction; neither `python2` nor `yarn` are used by micboard (so unless they're used by something else you can safely remove them again). And as to the `nvm` lines... they're forcing node back to version 10 again (if only for the current terminal instance, rather than globally).

The following, on a fresh install of debian, should be sufficient (I think).

```bash
sudo apt update
sudo apt-get install git nodejs npm python3-tornado
git clone --branch=0.8.7-updates https://github.com/karlcswanson/micboard.git
cd micboard
npm install --omit=dev
npm run build
```

Alternatively, to continue from where you left off - and in a *new* terminal window, please:

```bash
node --version
# The response to this should be something like "v18.19.0". If it starts "v10." then something has gone wrong.

sudo apt-get install python3-tornado

# I'm assuming you're in the "micboard" folder
git checkout 0.8.7-updates
npm install --omit=dev
npm run build
```

If that all works without errors, then run the micboard server with `python py/micboard.py`, open a web-browser, and navigate to `127.0.0.1:8058`.

**Reply by gunner4565** [reply]
Wow.  Thanks for the helpful info.  I'll try it tonight and let you know.  
Jerry

**Reply by gunner4565** [reply]
I had a few minutes this morning.  Thanks for all your help!!  It's now working.  I'll set it up later tonight and let you know how it goes.
Jerry
